### ISSUE TITLE
[bug] Hostname fails to parse in initialization if it has trailing /

### ISSUE THREAD

**Issue by rdevon** [facebookresearch/visdom#527]
Trying to connect to my visdom server now returns this error client side:

```
ERROR:visdom:Handshake status 200 OK
```

On the server side, I get:

```
ERROR:tornado.application:Uncaught exception POST /:<PORT>/env/main (<IP>)
HTTPServerRequest(protocol='http', host=<IP>, method='POST', uri='/:<PORT>/env/main', version='HTTP/1.1', remote_ip=<IP>)
Traceback (most recent call last):
  File "/home/devonh/.local/lib/python2.7/site-packages/tornado/web.py", line 1590, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/visdom/server.py", line 1086, in post
    username = json_obj["username"]
KeyError: u'username'
ERROR:root:ERROR: 500: {'exc_info': (<type 'exceptions.KeyError'>, KeyError(u'username',), <traceback object at 0x7fa292433488>)}
```

visdom on client and server side are both installed via:

```
pip install git+https://github.com/facebookresearch/visdom.git --user --upgrade
```
So version is visdom-0.1.8.5

I tried upgrading tornado as well, but it's up to date: tornado-5.1.1

**Comment by JackUrb** [comment]
What call is causing this to happen, just instantiating `visdom.Visdom()`? This is a strange error as it points to you making a post request to the `IndexHandler`, which is a fallback if nothing else hits in the path pattern-matching. I'm betting that the issue is the address it's pointing to is **`/:<PORT>/`**`env/main` rather than `/env/main` (<IP>:<PORT>). How are you passing the arguments to the `visdom.Visdom()` instance?

**Comment by rdevon** [comment]
Thank you for the fast response.

Here is the python:
```
logger.info('Using visdom version {}'.format(visdom.__version__))
print(server, port)
visualizer = visdom.Visdom(server=server, port=port)
print(visualizer.check_connection())
```

and the output:
```
Using visdom version 0.1.8.5
http://<IP>/ <PORT>
WARNING:root:Setting up a new session...
ERROR:visdom:Handshake status 200 OK
ERROR:visdom:Handshake status 200 OK
WARNING:visdom:Visdom python client failed to establish socket to get messages from the server. This feature is optional and can be disabled by initializing Visdom with `use_incoming_socket=False`, which will prevent waiting for this request to timeout.
ERROR:visdom:Handshake status 200 OK
False
```

**Comment by JackUrb** [comment]
Are you including a `/` in your server address?

**Comment by rdevon** [comment]
It's in the format:
http://X.X.X.X/

**Comment by JackUrb** [comment]
Try dropping the `/` at the end. I think this might be a silly bug that visdom should probably handle better.

**Comment by rdevon** [comment]
Ahhhh, yes that's it. Yeh, not clear on either the server or client side.

Feel free to close if you think this is good for now.

**Comment by JackUrb** [comment]
Fixed in #531
